microproxy/proxy.py

```diff
@@ -86,28 +87,28 @@ def next_layer(self, current_layer, context):
 
 
 class ProxyServer(tcpserver.TCPServer):
-    def __init__(self, config, proxy_server_handler=None):
-        super(ProxyServer, self).__init__()
+    def __init__(self, config, **kwargs):
+        super(ProxyServer, self).__init__(**kwargs)
         self.config = config
-        self.host = config["host"]
-        self.port = config["port"]
         self.interceptor = Interceptor(config=config)
+        self.layer_manager = LayerManager(config)
```

Comment:
We need to decide using a single instance of layer_manager or create each one when handle_server. Currently, there will be no problem either way.